### ISSUE TITLE
Canonical way to delete "env" variable?

### ISSUE THREAD

**Issue by KelSolaar** [nerdvegas/rez#875]
Hi,

I'm trying to delete a variable in an `env` attribute in my `commands` definition and I'm having a hard time. Any idea?

Cheers,

Thomas

**Comment by KelSolaar** [comment]
So it seems like `unsetenv` does work but not quite fully. I just lost 2 hours on that...

```python
    print(env.OCIO_ACTIVE_DISPLAYS)
    unsetenv('OCIO_ACTIVE_DISPLAYS')
    print(env.OCIO_ACTIVE_DISPLAYS)
```

```
Display 1:Display 2
Display 1:Display 2
```

But then in Maya:

```python
import os
print os.environ['OCIO_ACTIVE_DISPLAYS']
# Error: OCIO_ACTIVE_DISPLAYS
# Traceback (most recent call last):
#   File "<maya console>", line 2, in <module>
#   File "[...]maya\bin\python27.zip\os.py", line 425, in __getitem__
#     return self.data[key.upper()]
# KeyError: 'OCIO_ACTIVE_DISPLAYS' # 
```

So it does unset the variable but it is partially cache somewhere still...

**Comment by nerdvegas** [comment]
Looks like a bug to me. Let's ticket it up.
Thx
A

**Comment by KelSolaar** [comment]
It really drove me nuts yesterday, went through the code (rez\utils\data_utils.py, rez\rex.py) which is really hard to follow with multiple nested data structure referencing each others. It would certainly benefit from documentation and a solid battery of unit tests. Thing that did not help is that I only found `unsetenv` very late too, might be worth referencing directly around here: https://github.com/nerdvegas/rez/wiki/Package-Commands#variable-appending-and-prepending and not at the very end.
